osv/purl_helpers.py

```diff
@@ -18,23 +18,44 @@
 
 from packageurl import PackageURL
 
-PURL_ECOSYSTEMS = {
-    'Alpine': 'apk',
-    'Bitnami': 'bitnami',
-    'crates.io': 'cargo',
-    'Debian': 'deb',
-    'Go': 'golang',
-    'Hackage': 'hackage',
-    'Hex': 'hex',
-    'Maven': 'maven',
-    'npm': 'npm',
-    'NuGet': 'nuget',
-    'OSS-Fuzz': 'generic',
-    'Packagist': 'composer',
-    'Pub': 'pub',
-    'PyPI': 'pypi',
-    'RubyGems': 'gem',
-    'SwiftURL': 'swift',
+# PURL spec:
+# https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst
+ECOSYSTEM_PURL_DATA = {
+    'AlmaLinux': ('rpm', 'almalinux'),
+    'Alpine': ('apk', 'alpine'),
+    # Android
+    'Bitnami': ('bitnami', None),
+    'Chainguard': ('apk', 'chainguard'),
+    'CRAN': ('cran', None),
+    'crates.io': ('cargo', None),
+    'Debian': ('deb', 'debian'),
+    # GIT
+    'GitHub Actions': ('github', None),
+    'Go': ('golang', None),
+    'Hackage': ('hackage', None),
+    'Hex': ('hex', None),
+    # Linux
+    'Maven': ('maven', None),
+    'npm': ('npm', None),
+    'NuGet': ('nuget', None),
+    'openSUSE': ('rpm', 'opensuse'),
+    'OSS-Fuzz': ('generic', None),
+    'Packagist': ('composer', None),
+    'Pub': ('pub', None),
+    'PyPI': ('pypi', None),
+    'Red Hat': ('rpm', 'redhat'),
+    'Rocky Linux': ('rpm', 'rocky-linux'),
+    'RubyGems': ('gem', None),
+    'SUSE': ('rpm', 'suse'),
+    'SwiftURL': ('swift', None),
+    'Ubuntu': ('deb', 'ubuntu'),
+    'Wolfi': ('apk', 'wolfi'),
+}
+
+# Create the reverse lookup hash map
+PURL_ECOSYSTEM_MAP = {
+    (purl_type, purl_namespace): ecosystem
```

Comment:
Same here, use PackageURL as the key if possible, otherwise use namedtuple as key.

Reply:
Updated with a namedtuple